misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/IterableAssertionsSpec.kt

```diff
@@ -126,4 +128,18 @@ abstract class IterableAssertionsSpec(
         }
     }
 
+
+    describeFun(containsNoDuplicates) {
+        val containsNoDuplicatesFun = containsNoDuplicates.lambda
+
+        it("list without duplicates") {
+            expect(listOf(1, 2) as Iterable<Int>).containsNoDuplicatesFun()
+        }
+
+        it("list with duplicates") {
+            expect {
+                expect(listOf(1, 2, 1, 2, 3, 4, 4, 4) as Iterable<Int>).containsNoDuplicatesFun()
+            }.toThrow<AssertionError> { messageContains("$hasDescriptionBasic: $duplicateElements") }
+        }
```

Comment:
Please also check that it reports the elements as expected. In your example all
elements including position with the expection of 3 should be in reporting

Reply:
not sure if what you expect but implement that